### PR TITLE
Add test 'Sort and pagination shopping carts'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Add test 'Sort and pagination shopping carts'
| Type?         | refacto
| Category?     | TE
| BC breaks?    |no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/02_orders/05_shoppingCarts/02_sortAndPagination.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21255)

### CODE REVIEW

File: tests/UI/campaigns/data/demo/shoppingCarts.js [hunk 1]
```diff
@@ -3,35 +3,35 @@ module.exports = {
     {
       id: 1,
       orderID: 1,
-      customer: 'J. DOE',
+      customer: 'DOE',
```
Comment (SimonGrn) [inline tests/UI/campaigns/data/demo/shoppingCarts.js:6]: You're not breaking another test with this change ? No test uses this `demo/shoppingCarts` file ?

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/01_filterShoppingCarts.js [hunk 2]
```diff
@@ -101,7 +101,7 @@ describe('Filter the Shopping carts table', async () => {
     {
       args:
         {
-          testIdentifier: 'filterCarrier',
+          testIdentifier: 'filterCustomer',
```
Comment (SimonGrn) [inline tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/01_filterShoppingCarts.js:133]: Nice catch

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/02_sortAndPagination.js [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+// Import expect from chai
+const {expect} = require('chai');
+
+let browserContext;
+let page;
+let numberOfCustomers = 0;
+
+/*
+
+*/
```
Comment (SimonGrn) [inline tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/02_sortAndPagination.js]: Missing description

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/02_sortAndPagination.js [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+        args: {
+          testIdentifier: 'sortByOrderIDAsc', sortBy: 'status', sortDirection: 'up',
+        },
+      },
+      {
+        args: {
+          testIdentifier: 'sortByOrderIDDesc', sortBy: 'status', sortDirection: 'down',
+        },
+      },
+      /* Sort by carrier not working, skipping it
```
Comment (SimonGrn) [inline tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/02_sortAndPagination.js:190]: We need to keep track of this somewhere

**Review by SimonGrn — COMMENTED** [review]
A few comments

File: tests/UI/campaigns/data/demo/shoppingCarts.js (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/data/demo/shoppingCarts.js:6]: No test :)

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/01_filterShoppingCarts.js (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/01_filterShoppingCarts.js:133]: thx :)

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/02_sortAndPagination.js (hunk 3 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/02_sortAndPagination.js]: Done tnx :)

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/02_sortAndPagination.js (hunk 4 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/02_sortAndPagination.js:190]: Done :)

**Review by NeOMakinG — DISMISSED** [review]

**Comment by NeOMakinG** [comment]
@nesrineabdmouleh You've some conflict to resolve !

**Review by Progi1984 — DISMISSED** [review]

**Review by SimonGrn — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Comment by NeOMakinG** [comment]
Travis is red, but looks not related to this

**Comment by boubkerbribri** [comment]
> Travis is red, but looks not related to this

It's weird that unit tests are failing, I did restart the jobs.

**Pull request merged**